proposed/2023/allow-version-property-in-projectreference.md

```diff
@@ -0,0 +1,201 @@
+# Allow users to define version ranges for ProjectReferences
+
+- [Martin Ruiz](https://github.com/martinrrm)
+- Start Date (2023-01-01)
+- [5556](https://github.com/NuGet/Home/issues/5556)
+
+# Summary
+
+Add a `Version` to `ProjectReference` tag in CSPROJ, to allow customers to specify the referenced project version in the `.nupkg` and `nuspec` files when doing a pack command.
```

Comment:
For some reason, I think `Version` attribute doesn't belong to `ProjectReference` because it refers to the package version range. Looking at the schema for `ProjectReference` element https://github.com/dotnet/msbuild/blob/main/src/MSBuild/MSBuild/Microsoft.Build.CommonTypes.xsd#L643-L722, I think `PackageVersion` might be more appropriate. The schema has `Package` as sub element for `ProjectReference` but there are no comments to understand more about its usage. I noticed that there is also `SpecificVersion` attribute for `ProjectReference` element to specify whether the exact version of the assembly should be used.

Reply:
FYI - There is an MSBuild property `PackageVersion` allowing customers to specify package version in the csproj file if the project is packed as nupkg.

https://learn.microsoft.com/nuget/create-packages/package-authoring-best-practices#package-metadata

Visual Studio property name | Project file/ MSBuild property name | Nuspec property name | Description
-- | -- | -- | --
Package version | PackageVersion | version | NuGet package version.